### PR TITLE
Adds a functionality to link a dataset to another.

### PR DESCRIPTION
**Pull request by ASalvail**
Keeps the internals (symbolic parts) the same.

### CODE REVIEW

**Comment by MarcCote** [comment]
Just so we have a summary of our discussion can you add a small docstring saying in what occasion  one would uses such method. After that, I'll it the merge button.

File: smartlearner/interfaces/dataset.py [hunk 1]
```diff
@@ -94,3 +94,15 @@ def target_size(self):
 
     def __len__(self):
         return len(self.inputs.get_value())
+
+    def create_linked_dataset(self, inputs, targets=None, name=None, keep_on_cpu=None):
+        name = name if name is not None else self.name + '_linked'
+        koc = keep_on_cpu if keep_on_cpu is not None else self.keep_on_cpu
+
+        dset = Dataset(inputs, targets, name, koc)
```
Comment (ppoulin91) [inline smartlearner/interfaces/dataset.py]: Fix for subclass usage

**Comment by ASalvail** [comment]
Should be ready for merge.

**Comment by MarcCote** [comment]
Thanks @ASalvail

**Pull request merged**